RecoLocalCalo/HcalRecAlgos/src/ZdcSimpleRecAlgo_Run3.cc

```diff
@@ -0,0 +1,547 @@
+#include "RecoLocalCalo/HcalRecAlgos/interface/ZdcSimpleRecAlgo_Run3.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "CalibCalorimetry/HcalAlgos/interface/HcalTimeSlew.h"
+#include <algorithm>  // for "max"
+#include <iostream>
+#include <cmath>
+
+#include "DataFormats/METReco/interface/HcalCaloFlagLabels.h"
+
+constexpr double MaximumFractionalError = 0.0005;  // 0.05% error allowed from this source
+
+ZdcSimpleRecAlgo_Run3::ZdcSimpleRecAlgo_Run3(bool correctForTimeslew,
+                                             bool correctForPulse,
+                                             float phaseNS,
+                                             int recoMethod)
+    : recoMethod_(recoMethod),
+      correctForTimeslew_(correctForTimeslew),
+      correctForPulse_(correctForPulse),
+      phaseNS_(phaseNS) {}
+
+ZdcSimpleRecAlgo_Run3::ZdcSimpleRecAlgo_Run3(int recoMethod) : recoMethod_(recoMethod), correctForTimeslew_(false) {}
+
+void ZdcSimpleRecAlgo_Run3::initPulseCorr(int toadd, const HcalTimeSlew* hcalTimeSlew_delay) {
+  if (correctForPulse_) {
+    pulseCorr_ = std::make_unique<HcalPulseContainmentCorrection>(
+        toadd, phaseNS_, false, MaximumFractionalError, hcalTimeSlew_delay);
+  }
+}
+
+// helper functions for pedestal subtraction and noise calculations
+namespace zdchelper {
+  inline double subPedestal(const float energy, const float ped, const float width) {
+    if (energy - ped > width)
+      return (energy - ped);
+    else
+      return (0);
+  }
+
+  // assumes that if noise TS is from pileup, noise TS should be
+  // about 40% of previous energy and only 40% of noiseTS will be subtracted from signal
+  inline double calcNoise(const float energy0, const float energy1) {
+    // if(energy0<=0 || energy1<=0)return (energy1);
+    // // check if noise TS is less than half of previous TS
+    // else if( energy0 >= 2*energy1) return( .4*energy1);
+    // else return(.4*energy1);
+    return (.4 * energy1);
+  }
+  inline double calcNoiseRPD(const float energy0, const float energy1) {
```

Comment:
If this function is not using `energy0` then don't pass it. 
Please remove commented code here and elsewhere, unless there's a good reason to keep it around.

Reply:
If the RecHit Algo does not use energy0, it will be removed. We are finalizing an updated energy and noise subtraction and once updated, there should not be any unused variables.

Reply:
The code for the noise subtraction has been updated and energy0 is in use now